crates/metrics/src/lib.rs

```diff
@@ -0,0 +1,110 @@
+use jemalloc_ctl::{epoch, epoch_mib, stats, thread};
+
+#[global_allocator]
+static ALLOC: jemallocator::Jemalloc = jemallocator::Jemalloc;
+
+/// Statistics related to memory allocations for the entire (rust portion) of the application. The
+/// precise meaning of each field is included in the [jemalloc man
+/// page](http://jemalloc.net/jemalloc.3.html). Each field in this struct can be found in the man
+/// page prefixed by "stats.".
+#[derive(Debug)]
+#[repr(C)]
+pub struct GlobalMemoryStats {
+    pub active: u64,
```

Comment:
I'd expect this to be separately tracking allocated _bytes_ vs allocated _objects_. Both have interesting costs.

Maybe it is already, but it's not clear to me from the naming. And if it's not (currently) possible to have both, I think the names should still clarify which is measured.

Reply:
jemalloc seems to only expose stats for bytes, not actual counts of allocations. We could get both by further instrumenting the jemalloc allocator using the `stats_alloc` crate to track individual calls to alloc and dealloc, but I wasn't really sure if that was worth it. It'd be easy to add if you think it's worth the runtime overhead (basically just an atomic increment for every call).

As far as the names go, the names that are there are meant to match the names used by jemalloc in their docs. IMO it's better to use their names exactly, since these are all very specific to jemalloc, and are already well documented.